### PR TITLE
Harmonise the CLIs and create a unified CLI for PM and controller

### PR DESCRIPTION
**Pull request by plasorak**

### CODE REVIEW

**Comment by plasorak** [comment]
Change in this PR (note none of the changes in this PR are on the server-side, so endpoint behaviour is unchanged):
 - Quite a bit of infrastructure work:
   - Controllers now use a driver similar to what the PM uses (this simplifies a bit the CLI code).
   - PM now uses better exception handling (more or less derived from the controller, but still needs a bit of work - I'm not very fond of gRPC Python exceptions).
   - Each command is now able to have a `--traceback` option that prints the full stack upon request (for the controller, this means printing the stack on the remote server in certain cases, for the PM, not quite yet)
   - Separation of command, shell and context in controller and PM
   - I've put the common utilities in utils
   - There is now a `connect` command inside the controller, to connect to a different controller
- Unified CLI that is able to talk to the controller and the process manager. With the above changes, this became trivial.

File: src/drunc/apps/__main_controller_shell__.py [hunk 1]
```diff
@@ -1,12 +1,21 @@
...
-    except Exception:
-        from drunc.utils.utils import print_traceback
-        print_traceback()
+
+    except Exception as e:
+        if context.print_traceback:
+            from drunc.utils.utils import print_traceback
+            print_traceback()
+        from rich import print as rprint
+        rprint(f'[red bold]:fire::fire: Exception thrown :fire::fire:\n[/][yellow]{e}[/]\nUse [blue]--traceback[/] for more information\nExiting...')
```
Comment (gcrone) [inline src/drunc/apps/__main_controller_shell__.py:15]: Should the advice to use `--traceback` only be printed in an else clause here? Or maybe the `print_traceback` function could take the `context.print_traceback` as an argument and the `if` could be internal.

Comment (plasorak) [reply src/drunc/apps/__main_controller_shell__.py:15]: Definitely, this is a bit sloppy

Comment (plasorak) [reply src/drunc/apps/__main_controller_shell__.py]: Just corrected it everywhere

File: src/drunc/controller/interface/shell_utils.py [hunk 2]
```diff
@@ -1,3 +1,92 @@
...
+                sleep(0.5)
+
+        except Exception as e:
+            ctx.critical('Could not get the controller\'s status')
+            ctx.critical(e)
+            ctx.critical('Exiting.')
+            ctx.terminate()
+            raise e
+
+        else:
```
Comment (gcrone) [inline src/drunc/controller/interface/shell_utils.py:64]: I can't see what `if` this `else` does with

File: src/drunc/process_manager/interface/commands.py [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+        rethrow = traceback,
+    )
+    async for result in results:
+        if not result: break
+        obj.print(f'\'{result.process_description.metadata.name}\' ({result.uuid.uuid}) process started')
+
+    controller_address = obj.get_driver('process_manager').controller_address
+    if controller_address:
+        from rich.panel import Panel
+        obj.print(Panel("Controller endpoint: 'np04-srv-019:3338', point your 'drunc-controller-shell' to it.", padding=(2,6), style='violet', border_style='violet'), justify='center')
```
Comment (gcrone) [inline src/drunc/process_manager/interface/commands.py:38]: Hard-wired `np04-srv-019:3338`? Shouldn't this be `controller_address`?

Comment (plasorak) [reply src/drunc/process_manager/interface/commands.py]: done

**Pull request merged**